InnerSource pattern "Creating an InnerSource Strategy" (#652)
gruetter/innerSourceTest: patterns/1-initial/creating_an_innersource_strategy.md

## Title

Creating an InnerSource Strategy

## Patlet / Summary

Sometimes, it is difficult to convince people of the relevance of InnerSource for your organization and/or to get support from management. Creating an InnerSource strategy, that connects your InnerSource approach and activities to the goals and the overall strategy of your organization, can help in this regard.

## Problem

- Despite an organization has an InnerSource program, it is challenging to establish InnerSource in that organization
- Missing management support and low awareness in some or many development teams are among the reasons
- Potentially, the InnerSource program itself is missing clear goals and/or approaches how to achieve those.

## Context

> - InnerSource is not an end in itself. It must deliver clear benefits to your organization and support its goals

Reviewer: This reads more like a general statement about InnerSource, than like Context of a specific org. I think you could leave away this bullet entirely without negatively impacting the content.

> - If InnerSource and its benefits is mainly explained and positioned in a rather abstract way, it might be difficult to convince management and a critical mass of the development teams

Reviewer: If I were to rewrite this as "context of an organization that has this challenge", I might write it like this:

```suggestion
- The program office at the organization has expressed an ambition to follow InnerSource practices throughout the organization. However InnerSource and its benefits have been explained in a rather abstract way, making it difficult to convince management and a critical mass of the development teams.
```

- People see InnerSource as a rather abstract methodology which does not really fit to their situation. They might say/think "InnerSource sounds great, but it is not relevant for our organization"
- People struggle to make a connection between InnerSource and their goals/the goals of their organisation

## Forces

> TODO

Reviewer: Porting these potential forces here from the current "Problem" section.

Are these good "constraints that can be changed at a cost"? Is the solution that you are proposing "changing one or more of these forces in order to solve the problem"?

I am often confused myself between what goes into the **Context** and **Forces** section, party because these ideas from a rather scientific context. So I cannot blame you if you are confused here as well :)

```suggestion
- The InnerSource program itself is missing clear goals and/or approaches how to achieve those
- Missing management support for the InnerSource program
- Low awareness of the InnerSource program in some development teams
```

## Title

Creating an InnerSource Strategy

## Patlet / Summary

Sometimes, it is difficult to convince people of the relevance of InnerSource for your organization and/or to get support from management. Creating an InnerSource strategy, that connects your InnerSource approach and activities to the goals and the overall strategy of your organization, can help in this regard.

## Problem

Despite an organization has an InnerSource program, it is challenging to establish InnerSource in that organization. Missing management support and low awareness in some or many development teams are among the reasons. Potentially, the InnerSource program itself is missing clear goals and/or approaches how to achieve those.

> ## Context

Reviewer: The content for the "Context" section should focus on this:

> Where does the problem exist? What are the pre-conditions? Unchangeable before the solution goes into place. The content here is often tied to the applicability of the pattern for other readers: "Do I have this same particular situation?"

What can we write here about the given context at an org where this problem exists?

Minor: In terms of formatting we recommend to use bullets, so that one can separate distinct points more clearly.

Author: I turned that paragraph into a list of bullet points, but did not change the content since I am not sure what is required here. At least on a high level, the paragraph explains where the problem exists, IMO. It could be described in further detail. Is that what you mean or would you expect different content?

Reviewer: @misappi we misunderstood each other here. I meant the "Context" section, not the "Problem" section.

Author: OK, got it. I extended the context paragraph and converted it into a list of bullet points.

Reviewer: @misappi left some comments related to the Context/Forces inline.

Do you want to revert the Problem section to its previous state? Typically we just use prose and no bullets there.

> InnerSource is not a means to an end. It must deliver clear benefits to your organization and support its goals. If InnerSource and its benefits is mainly explained and positioned in a rather abstract way, it might be difficult to convince management and a critical mass of the development teams. People might say/think "InnerSource sounds great, but it is not relevant for our organization".

Reviewer 2: @misappi , did you intend to say here that InnerSource in not an _end_ in itself, it is a mean to an end?

Author: "... not an end in itself ..." -> corrected

## Forces

TODO

<!--
What makes the problem difficult?
What are the trade-offs?
These are constraints that **can be changed** at a cost.
The solution might change one or more of these forces in order to solve the problem, while also in-turn changing the context.
-->

## Solutions

Create an InnerSource strategy. Such a document describes how the application of InnerSource relates to the goals and business strategy of your organization. It explains why InnerSource is important (not only in an abstract way but specifically to your organization), why you should leverage it, what goals your organization should have with respect to InnerSource, and what actions to perform to achieve those.

A high level table of contents of an InnerSource strategy document could look like so:

- Executive summary
- What InnerSource is
- Why InnerSource is important for our organization

> - Current status of InnerSource in your organization

Reviewer: "our" or "your"?

My preference is "our", as this seems to be written from the perspective of the org that is creating the InnerSource strategy document.

Author: changed

If your organization has an approval process for strategy document, it is recommended to get an approval for your InnerSource strategy. Such a process can trigger discussions about InnerSource in your organization, raise awareness and make such a strategy document more "official".

## Resulting Context

TODO

<!--
What is the situation after the problem has been solved?
The original context is changed indirectly by way of the solution.
Often this section can include discussion of the next possible Patterns/problems introduced.
This section can be short in content - the solution may not introduce new problems or change much context.
-->

> ## Known Instances (optional)

Reviewer: ```suggestion
## Known Instances
```

> SAP has an InnerSource strategy (see blog post "[Cultivating InnerSource at SAP](https://community.sap.com/t5/open-source-blogs/cultivating-innersource-at-sap/ba-p/13575274)")

Reviewer: ```suggestion
- **SAP** has an InnerSource strategy (see blog post "[Cultivating InnerSource at SAP](https://community.sap.com/t5/open-source-blogs/cultivating-innersource-at-sap/ba-p/13575274)")
```

changing formatting

> 
> TODO
> 
> <!--
> Where has this been seen before?
> Helps to reinforce that this is a REAL pattern and that you match the context.
> 
> May mention:
> 
> * A particular business
> * Anonymized instances ex: "3 companies have proven that this is a good solution" or "A large financial services org...".
> -->

Reviewer: ```suggestion
```

For status "initial" a known instance isn't even required. So even having the SAP example in here is already amazing. Therefore I recommend to remove the TODO+comments here.

## Status

Initial

> ## Author(s) (optional)

Reviewer: ```suggestion
## Author(s)
```

> Michael Picht

Reviewer: ```suggestion
- Michael Picht
```

making room for additional authors, should we get further significant contributions to this pattern
